controller/service.go

```diff
@@ -109,6 +111,8 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 		// User set or changed the desired LB IP, nuke the
 		// state. allocateIP will pay attention to LoadBalancerIP and try
 		// to meet the user's demands.
+		// TODO: remove check for svc.Spec.LoadBalancerIP when this field is removed
+		// from service spec.
```

Comment:
since the annotation takes precedence over loadbalancerIp then there is no need to do both in this case and overwrite with annotation u can do the annotation 1st and skip loadbalancer check if len(desiredLbIps) > 0

Reply:
Good find ! changed it.